lib/dry/transaction/sequence.rb

```diff
@@ -231,12 +237,14 @@ def append(other = nil, **options, &block)
       # @api public
       def insert(other = nil, before: nil, after: nil, **options, &block)
         insertion_step = before || after
-        unless steps.map(&:step_name).include?(insertion_step)
+        match_insertion_step = proc { |step| step.step_name == insertion_step }
+
+        unless steps.any?(&match_insertion_step)
```

Comment:
Just to clarify this, when I saw the line I noticed it can be made faster so I made a very quick and dirty benchmark:
```ruby
require "benchmark"
require 'bundler/setup'
require "dry-transaction"

SA = Dry::Transaction::StepAdapters

class Foo
  attr_reader :name
  def initialize(name)
    @name = name
  end
end

collection = (1..10).map { |n| Dry::Transaction::Step.new(SA[:step], :"name#{n}", "operation_name", -> {}, {}) }

names = (1..11).map { |n| :"name#{n}" }

Benchmark.bmbm do |x|
  x.report("map(...).include?(...)") { 1_000_000.times { |i| name = names.sample; collection.map(&:step_name).include?(name) } }
  x.report("any? { ... }") { 1_000_000.times { |i| name = names.sample; collection.any? { |e| e.step_name == name } } }
end
```
Result:
```
Rehearsal ----------------------------------------------------------
map(...).include?(...)   1.390000   0.060000   1.450000 (  1.468051)
any? { ... }             0.570000   0.010000   0.580000 (  0.607362)
------------------------------------------------- total: 2.030000sec

                             user     system      total        real
map(...).include?(...)   1.220000   0.040000   1.260000 (  1.277542)
any? { ... }             0.590000   0.000000   0.590000 (  0.598434)
```

Seems like it's twice as fast. Now I know this code is not a bottleneck and this is a very very small improvement but since dry-rb emphasises speed I thought it might still be worthwhile since it seems to me that there is no loss of readability. In my experience these kind of very small wins end up making a difference when there's enough of them.